src/features/ValueEditor/ValueEditor.tsx

```diff
@@ -22,10 +21,69 @@ import {
   FiPlus,
 } from 'react-icons/fi'
 import {LuEraser} from 'react-icons/lu'
-import {Validation} from '../../types'
+import {ZodSchema, z} from 'zod'
+import {generateErrorMessage} from 'zod-error'
 import {CopyButton} from '../CopyButton'
 import classes from './ValueEditor.module.css'
 
+const evaluateExpression = (expression: string) => {
+  try {
+    const evaluatedExpression = new Function(`return ${expression}`)()
+    return {success: true, data: evaluatedExpression} as const
+  } catch (e) {
```

Comment:
Avoid the sequence of ternary operators, also would put `catch {` on a newline (dunno if prettier forces that style).
Btw:
```
  catch (e) {
    let error;
    if (e instanceof SyntaxError) error = 'Invalid syntax'
    else if (e instanceof ReferenceError) error = 'Invalid reference'
    else error = 'Invalid value'

    return {success: false, error} as const
  }
```

Reply:
Removed nested ternary operators even though I'm not against them. They offer concise syntax and are even readable with the right indentation